src/server/rdb_load.cc

```diff
@@ -2467,6 +2467,11 @@ void RdbLoader::LoadItemsBuffer(DbIndex db_ind, const ItemsBuf& ib) {
       stop_early_ = true;
       break;
     }
+    if (!pv.TagAllowsEmptyValue() && pv.Size() == 0) {
+      LOG(ERROR) << "Found empty key: " << item->key << " in DB " << db_ind << " rdb_type "
+                 << item->val.rdb_type;
+      continue;
+    }
```

Comment:
Did you try replicating from our previous version with empry hsets? The fail is triggered _before_ your code in FromOpaque, so it won't help

Reply:
That's actually a good suggestion -- let's make it a test 👍